src/Bicep.LangServer/Completions/SyntaxMatcher.cs

```diff
@@ -90,11 +90,24 @@ public static List<SyntaxBase> FindNodesMatchingOffset(ProgramSyntax syntax, int
             {
                 // callback is invoked only if node span contains the offset
                 // in inclusive mode, 2 nodes can be returned if cursor is between end of one node and beginning of another
-                // we will pick the node to the left as the winner
-                if (nodes.Any() == false || TextSpan.AreNeighbors(nodes.Last(), current) == false)
+                // we will pick the node to the left as the winner if it's not a newline. Otherwise, pick the right one instead.
+                if (!nodes.Any())
```

Comment:
This is basically an optimization for
```bicep
|var foo = true
```
that changes the matching nodes from
```
[VariableDeclarationSytax, Token (NewLine)]
```
to
```
[VariableDeclarationSytax, IdentifierSyntax, Token (Identifier)]
```

Reply:
Worth adding the explanation as a comment?